### PR TITLE
Remove use of std::mutex in gtest code.

### PR DESCRIPTION
**Pull request by dneto0**
Gtest runs in single-threaded mode.  So the GlslangInitializer
object in the test code doesn't have to do its own synchronization.

Credit to @AWoloszyn and @antiagainst for suggesting this approach and determining this is safe to do.

### CODE REVIEW

**Comment by dneto0** [comment]
PTAL @AWoloszyn @antiagainst

**Comment by antiagainst** [comment]
LGTM

**Comment by dneto0** [comment]
PTAL @johnkslang

File: gtests/Initializer.h [hunk 1]
```diff
@@ -63,55 +61,25 @@ class GlslangInitializer {
...
     // Re-initializes glsl state iff the previous messages and the current
-    // messages are incompatible.
+    // messages are incompatible.  We assume external synchronization, i.e.
+    // there is at most one acquired token at any one time.
     InitializationToken acquire(EShMessages new_messages)
     {
-        stateLock.lock();
-
         if ((lastMessages ^ new_messages) &
             (EShMsgVulkanRules | EShMsgSpvRules)) {
```
Comment (johnkslang) [inline gtests/Initializer.h]: It seems this should be handled by proper symbol table caching/keying, which isn't done yet.  Not related to the PR, just noting it.

**Pull request merged**